packages/angular/src/migrations/update-12-3-0/normalize-angular-strict-defaults.ts

```diff
@@ -0,0 +1,79 @@
+import {
+  formatFiles,
+  readJson,
+  readWorkspaceConfiguration,
+  Tree,
+  updateWorkspaceConfiguration,
+  WorkspaceConfiguration,
+} from '@nrwl/devkit';
+
+export default async function normalizeAngularStrictDefaults(host: Tree) {
+  const wsConfig = readWorkspaceConfiguration(host);
+
+  let shouldUpdate = false;
+  if (shouldUpdateStrictDefault(host, wsConfig, '@nrwl/angular:application')) {
+    normalizeGeneratorStrictDefault(wsConfig, '@nrwl/angular:application');
+    shouldUpdate = true;
+  }
+  if (shouldUpdateStrictDefault(host, wsConfig, '@nrwl/angular:library')) {
+    normalizeGeneratorStrictDefault(wsConfig, '@nrwl/angular:library');
+    shouldUpdate = true;
+  }
+
+  if (shouldUpdate) {
+    updateWorkspaceConfiguration(host, wsConfig);
+    await formatFiles(host);
+  }
+}
+
+function shouldUpdateStrictDefault(
+  host: Tree,
+  wsConfig: WorkspaceConfiguration,
+  generator: string
+): boolean {
+  if (wsConfig.generators?.[generator]) {
+    if (wsConfig.generators[generator].strict !== undefined) {
+      return false;
+    }
+  }
+  const [collection, generatorName] = generator.split(':');
+  if (wsConfig.generators?.[collection]) {
+    if (wsConfig.generators[collection][generatorName]?.strict !== undefined) {
+      return false;
+    }
+  }
+  if (
+    Object.keys(wsConfig.generators ?? {}).some((key) =>
+      key.startsWith('@nrwl/angular')
+    )
+  ) {
+    return true;
+  }
+  // if the config is not present the workspace could still be using
+  // Angular with the defaults
+  const pkgJson = readJson(host, 'package.json');
+  return (
+    pkgJson.dependencies?.['@nrwl/angular'] ||
+    pkgJson.devDependencies?.['@nrwl/angular']
+  );
+}
+
+function normalizeGeneratorStrictDefault(
```

Comment:
Perhaps I am misunderstanding the intention of this code. 

Is this keeping strict mode off for existing workspaces?

Is this something we should allow the dev to override with a flag? Perhaps they want to migrate and use strict mode?

Reply:
This is leaving the workspace as is, so, if the workspace has no configuration for the `strict` option, it's because is using the current default `false`. Since the default is changing, the migration restores what the workspace is using, so it explicitly set the default to `false` to keep existing behavior. The way our migrations work, it's always optional to run any migration. Devs can remove migrations from the `migrations.json` file before running `ng migrate --run-migrations` if they don't want to run a specific migration, so no flag would be needed.

That said, I think you're making a good point though. I think regardless of the workspace being new or not, the idea now would be to have any new apps/libs using `strict` by default. So, we shouldn't have any migration at all and if the dev hasn't explicitly set the option to `false` it should get the default `true`.

Reply:
I'll go ahead and remove the migration. We won't touch whatever the workspace has configured, but the new default will apply if the `strict` option is not set in the workspace.

Reply:
I think so too, that way the developer will be getting the default experience. And they can always then manually set `strict` to `false` afterwards.